### PR TITLE
apps: upgrade metrics-server to 3.10.0

### PR DESCRIPTION
**Pull request by lucianvlad**
**What this PR does / why we need it**:
Upgrade metrics-server to 3.10.0
**Which issue this PR fixes** *(use the format `fixes #<issue number>(, fixes #<issue_number>, ...)` to automatically close the issue when PR gets merged)*: fixes #

**Public facing documentation PR** *(if applicable)*

**Special notes for reviewer**:

**Add a screenshot or an example to illustrate the proposed solution:**

**Checklist:**

- [x] Added relevant notes to [WIP-CHANGELOG.md](https://github.com/elastisys/compliantkubernetes-apps/blob/main/WIP-CHANGELOG.md)
- [x] Proper commit message prefix on all commits
- [ ] Updated the [public facing documentation](https://github.com/elastisys/compliantkubernetes)
- Is this changeset backwards compatible for existing clusters? Applying:
    - [x] is completely transparent, will not impact the workload in any way.
    - [ ] requires running a migration script.
    - [ ] will create noticeable cluster degradation.
          E.g. logs or metrics are not being collected or Kubernetes API server
          will not be responding while upgrading.
    - [ ] requires draining and/or replacing nodes.
    - [ ] will change any APIs.
          E.g. removes or changes any CK8S config options or Kubernetes APIs.
    - [ ] will break the cluster.
          I.e. full cluster migration is required.
- Chart checklist (pick exactly one):
    - [ ] I upgraded no Chart.
    - [x] I upgraded a Chart and determined that no migration steps are needed.
    - [ ] I upgraded a Chart and added [migration steps](https://github.com/elastisys/compliantkubernetes-apps/blob/main/migration).

**Pipeline config** *(if applicable)*
If you change some config options (e.g. add/rename variable or change the default value) you may need to update the config used by the pipeline in `pipeline/config`.

### CODE REVIEW

**Comment by crssnd** [comment]
any changes in metrics that could affects the alerts or dashboards?

any other changes that we should be aware?

File: WIP-CHANGELOG.md [hunk 1]
```diff
@@ -45,6 +45,7 @@
 - prometheus-operator to v0.63.0
 - grafana to v9.3.8
 - Upgraded Dex to v2.36.0, chart v0.14.1
+- metrics-server to v3.10.0
```
Comment (crssnd) [inline WIP-CHANGELOG.md:48]: please add separately the chart version and the app version

**Comment by crssnd** [comment]
do we want the `{{ .Values.kubernetesPSP }}` flag set in https://github.com/elastisys/compliantkubernetes-apps/blob/main/helmfile/values/metrics-server.yaml.gotmpl#L3 ?

**Review by Xartos — APPROVED** [review]
Other than cristis comment it looks good

File: WIP-CHANGELOG.md (hunk 1 above)
Comment (lucianvlad) [reply WIP-CHANGELOG.md]: Done

**Comment by lucianvlad** [comment]
> do we want the `{{ .Values.kubernetesPSP }}` flag set in https://github.com/elastisys/compliantkubernetes-apps/blob/main/helmfile/values/metrics-server.yaml.gotmpl#L3 ?

Added it now
Thx

**Review by crssnd — APPROVED** [review]

**Pull request merged**